Navassa

buha opened the pull request:
EDIT: I've marked in bold the project files that are relevant for review. The other files have large portions of code from the Linux driver and we would like to have this code as near to identical as possible between no-OS and Linux.
drivers/rf-transceiver/navassa/adrv9002.c
drivers/rf-transceiver/navassa/adrv9002.h
drivers/rf-transceiver/navassa/adrv9002_conv.c
drivers/rf-transceiver/navassa/adrv9002_init_data.c
**projects/adrv9001/Makefile
projects/adrv9001/src.mk**
projects/adrv9001/src/app/ORxGainTable.h
projects/adrv9001/src/app/RxGainTable.h
projects/adrv9001/src/app/TxAttenTable.h
**projects/adrv9001/src/app/app_iio.c
projects/adrv9001/src/app/app_iio.h**
**projects/adrv9001/src/app/headless.c**
projects/adrv9001/src/firmware/Navassa_EvaluationFw.h
projects/adrv9001/src/hal/adi_platform.h
projects/adrv9001/src/hal/adi_platform_types.h
**projects/adrv9001/src/hal/no_os_platform.c**
**projects/adrv9001/src/hal/parameters.h**

~~This currently only builds using make. To build it in XSDK one needs to edit the compiler defines using the SDK to match the ones in Makefile. Support for forwarding CFLAGS -D to XSDK without any user action is underway.~~  Support for forwarding CFLAGS -D to XSDK was added to master and this project successfully builds both from command line and with XSDK.

DAC_DMA_EXAMPLE - ADRV9002_RX2TX2 flag enabled.
![DAC_DMA_EXAMPLE - RX2TX2](https://user-images.githubusercontent.com/8933663/94256488-8bf8af80-ff32-11ea-8590-1229b88ef381.png)
![DAC_DMA_EXAMPLE - RX2TX2b](https://user-images.githubusercontent.com/8933663/94256491-8d29dc80-ff32-11ea-92b3-13b41c898fb4.png)

IIO - independent:
![IIO_SUPPORT](https://user-images.githubusercontent.com/8933663/94256492-8d29dc80-ff32-11ea-9ce3-9ebca4590d99.png)
IIO - ADRV9002_RX2TX2 flag enabled:
![IIO_SUPPORT - RX2TX2](https://user-images.githubusercontent.com/8933663/94256493-8dc27300-ff32-11ea-98c1-39d839f8f727.png)

buha commented on projects/adrv9001/src/app/adrv9002_conv.c:
```diff
@@ -0,0 +1,772 @@
+				       uint8_t field[][8], const bool tx,
+				       const int channel)
+{
+}
```
will remove the function entirely

buha commented on projects/adrv9001/src/app/adrv9002_conv.c:
```diff
@@ -0,0 +1,772 @@
+{
+	int ret;
+	struct adrv9002_chan *chan;
+	adi_adrv9001_SsiTestModeData_e test_data = ssi_type ==
```
I actually totally agree ! But this is part of code that is common with Linux driver and I'd keep them identical if possible.

buha commented on projects/adrv9001/src/app/headless.c:
```diff
@@ -0,0 +1,789 @@
+#define ADRV9001_BF_EQUAL(mask, value) ((value) == ((value) & (mask)))
+
+/* init data stuff */
+extern struct adi_adrv9001_SpiSettings *adrv9002_spi_settings_get(void);
```
Not for now. Since they are few functions, each used only once, I guess it's ok to get them via extern from .c to .c files.

buha commented on projects/adrv9001/src/app/headless.c:
```diff
@@ -0,0 +1,789 @@
+	/* Not a valid case. At least one channel should be enabled */
+	[0] = {0, 0},
+	/* tx2:0 rx2:0 tx1:0 rx1:1 */
+	[1] = {0x1BE400, 0},
```
I agree, these values are not nice, but again, I'd keep it identical with the Linux version.

buha commented:
V2:
- CMOS support including latest Linux commits
- implemented review comments so far

adrimbarean commented on projects/adrv9001/src/hal/no_os_platform.c:
```diff
@@ -0,0 +1,671 @@
+	ret = gpio_remove(phal->gpio_reset_n);
+	if (ret)
+		return ret;
+
```
If `ADRV9002_RX2TX2` should the `gpio_ssi_sync` GPIO be freed as well?

buha replied:
yes, good point

adrimbarean commented on projects/adrv9001/src/hal/no_os_platform.c:
```diff
@@ -0,0 +1,671 @@
+	adi_hal_Cfg_t *halCfg = NULL;
+
+	if (devHalCfg == NULL) {
+		halError = (int32_t)ADI_HAL_NULL_PTR;
+		return halError;
```
`halError` doesn't seem to have a point here. We can just return the error value directly. Same with the last statement.

buha replied:
true, i removed it

buha commented:
V3: 
- rebased onto master (autogenerated XSDK project now builds because of recent changes on master)
- moved files around (better distinguishing between drivers, hal, firmware and app)

buha commented:
V4:
- added astyle/doxygen/cppcheck relevant entries in ignorefiles
- changes related to @adrimbarean review

buha commented:
V5:
- updated API to latest version and performed necessary changes in project files

buha commented:
V6:
- Fixed a bunch of doxygen warnings

buha commented:
V7:
- fixed astyle
- fixed unstable API by removing the equivalent of a Linux commit that attempts to fix an API problem but introduces instability in adrv9001 init.

buha commented:
V8:
- separate commit for astyle/cppcheck/drivers etc. checks

buha commented:
V9:
- fix attempt for drivers build skip
